tools/zsync/Makefile

```diff
@@ -0,0 +1,26 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=zsync
+PKG_VERSION:=f0ea808bfd957d7589420585b3d25081e0040e6d
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://codeload.github.com/ahmedammar/zsync/tar.gz/$(PKG_VERSION)?
```

Comment:
I would prefer using a git+https:// download with a hash over an external service.

Reply:
Sorry I don't understand what you mean, could you please point me to an example where it is done this way?

Reply:
I think it means: "You should not use codeload.github.com" but some other source.

Reply:
local tarball basically.

There's no good solution. codeload changes the hashes from time to time (I've only seen it when using submodules) and local tarballs also change hashes in surprising ways. Many of the tarballs currently on the OpenWrt mirrors have wrong hashes. It causes no issues as it falls back to cloning the git repo (not a shallow clone oddly enough).

The latter is smaller as it uses .xz as an archive, usually.